tensorflow_probability/python/distributions/continuous_bernoulli.py

```diff
@@ -0,0 +1,454 @@
+           validate_args: Python `bool`, default `False`. When `True`
+           distribution parameters are checked for validity despite possibly
+           degrading runtime performance. When `False` invalid inputs may
+           silently render incorrect outputs.
+          allow_nan_stats: Python `bool`, default `True`. When `True`,
+            statistics (e.g., mean, mode, variance) use the value "`NaN`" to
+            indicate the result is undefined. When `False`, an exception is
+            raised if one or more of the statistic's batch members are
+            undefined.
+          name: Python `str` name prefixed to Ops created by this class.
+
+        Raises:
+          ValueError: If p and logits are passed, or if neither are passed.
+        """
+        parameters = dict(locals())
+        if (lams is None) == (logits is None):
+            raise ValueError("Must pass lams or logits, but not both.")
+        self._lims = lims
+        with tf.name_scope(name) as name:
+            self._lams = tensor_util.convert_nonref_to_tensor(
+                lams, dtype_hint=tf.float32, name="lams"
+            )
+            self._logits = tensor_util.convert_nonref_to_tensor(
+                logits, dtype_hint=tf.float32, name="logits"
+            )
+        super(ContinuousBernoulli, self).__init__(
+            dtype=dtype,
+            reparameterization_type=reparameterization.FULLY_REPARAMETERIZED,
+            validate_args=validate_args,
+            allow_nan_stats=allow_nan_stats,
+            parameters=parameters,
+            name=name,
+        )
+
+    @staticmethod
+    def _param_shapes(sample_shape):
+        return {"logits": tf.convert_to_tensor(sample_shape, dtype=tf.int32)}
+
+    @classmethod
+    def _params_event_ndims(cls):
+        return dict(logits=0, lams=0)
+
+    @property
+    def logits(self):
+        """Input argument `logits`."""
+        return self._logits
+
+    @property
+    def lams(self):
+        """Input argument `lams`."""
+        return self._lams
+
+    def _batch_shape_tensor(self):
+        x = self._lams if self._logits is None else self._logits
+        return tf.shape(x)
+
+    def _batch_shape(self):
+        x = self._lams if self._logits is None else self._logits
+        return x.shape
+
+    def _event_shape_tensor(self):
+        return tf.constant([], dtype=tf.int32)
+
+    def _event_shape(self):
+        return tf.TensorShape([])
+
+    def _sample_n(self, n, seed=None):
+        lams = self._lams_parameter_no_checks()
+        cut_lams = tf.where(
+            tf.logical_or(
+                tf.less(lams, self._lims[0]), tf.greater(lams, self._lims[1])
+            ),
+            lams,
+            self._lims[0] * tf.ones_like(lams),
+        )
+        new_shape = tf.concat([[n], tf.shape(lams)], 0)
+        uniform = tf.random.uniform(new_shape, seed=seed, dtype=lams.dtype)
+        sample = tf.where(
```

Comment:
Since this is used so often, I'd just recommend making a private method:

```
def not_within_bounds(lams):
   return lams < self._lims[0] | lams > self._lims[1]
```

and just use that everywhere.